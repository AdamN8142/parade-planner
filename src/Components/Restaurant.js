import React, { Component } from 'react';
import '../Restaurant.css';

class Restaurant extends Component {
  constructor() {
    super();
    this.state = {}
  }


  render() {
    return(
      
      this.props.sortedRestaurants.map(restaurant => {
        return (
          <div className="individual-restaurant">
            <img src={restaurant.img} />
            <a href={restaurant.website} target="_blank">{restaurant.name.toUpperCase()}</a>
            <p>#{restaurant.cuisine.join(' #')}</p>
            <p>{restaurant.rating}</p>
            <p>{restaurant.distance_to_parade}</p>
            <p>{restaurant.walk_to_parade}</p>
          </div> 
        )
      })
     
    )
  }
}
export default Restaurant