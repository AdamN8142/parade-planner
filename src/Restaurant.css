.individual-restaurant{
    background-color: goldenrod;
    height: 440px;
    margin-right: 7px;
    text-transform: capitalize;
    width: 500px;
    border-radius: 25px;
    box-shadow: 2px 5px purple;
    margin-bottom: 120px;
}

.individual-restaurant img {
    background-size: cover;
    margin-bottom: 10px;
    height: 191px;
    width: 300px;
    /* border:1px black solid; */
    border-radius: 25px;
    background-color: purple;
}

.themes {
    font-size: 16px;
    font-weight: bold;
    text-transform: lowercase;
}

.link {
    text-decoration: none;
    font-size: 20px;
    font-weight: bold;
    color: #720b98;
}

/* .img-and-name .rest-img {
    height: 150px;
    width: 200px;
} */

.restaurant-info {
    margin-left: 15px;
    font-family: monospace;
}


.rest-name {
    text-align: center;
    margin-top: 2px;
    margin-bottom: 2px;
    font-family: cursive;
    font-size: 16px;
    font-weight: bolder;
    padding:1px;
    text-decoration: overline;
}

.rest-tags{
    font-size: 12px;
    text-align: center;
    /* border-style: double; */
    padding:1px;
    border-color: darkred;
    border-radius: 45px;
    font-weight: bold;
    color: darkslateblue;
}