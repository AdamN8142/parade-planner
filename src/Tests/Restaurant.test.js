import React from "react";
import Restaurant from "../Components/Restaurant";
import { shallow } from "enzyme";

const mockRestaurants = [
  {
    "name": "Acme Oyster House",
    "location": "Metairie",
    "distance_to_parade": ".5 miles",
    "walk_to_parade": "10 minutes",
    "cuisine": ["seafood", "cajun", "creole", "classics"],
    "rating": 4.5,
    "img": "https://www.neworleansonline.com/images/slideshows/listings/2952/05.jpg",
    "website": "https://acmeoyster.com/metairie/"
  },
  {
    "name": "Zea Rotisserie & Bar",
    "location": "Metairie",
    "distance_to_parade": "1.7 miles",
    "walk_to_parade": "35 minutes",
    "cuisine": ["eclectic", "beer", "mediterranean", "teas"],
    "rating": 4.5,
    "img": "https://resizer.otstatic.com/v2/photos/wide-huge/24439769.jpg",
    "website": "http://zearestaurants.com/locations/metairie/"
  }];

describe("Restaurant", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Restaurant sortedRestaurants={mockRestaurants}
      />
    );
  });
	
  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });
});