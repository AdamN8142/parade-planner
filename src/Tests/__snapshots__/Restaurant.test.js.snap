// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant should match the snapshot with all data passed in 1`] = `
Array [
  <div
    className="individual-restaurant"
    key="0"
  >
    <img
      src="https://www.neworleansonline.com/images/slideshows/listings/2952/05.jpg"
    />
    <h5
      className="rest-name"
    >
      Acme Oyster House
    </h5>
    <p
      className="rest-tags"
    >
      #
      seafood #cajun #creole #classics
    </p>
    <div
      className="restaurant-info"
    >
      <p
        className="rest-lines"
      >
        <span
          className="bold"
        >
          Rating:
        </span>
         
        4.5
         / 5.0
      </p>
      <p
        className="rest-lines"
      >
        <span
          className="bold"
        >
          Distance:
        </span>
         
        .5 miles
      </p>
      <p
        className="rest-lines"
      >
        <span
          className="bold"
        >
          Walk-time:
        </span>
         
        10 minutes
      </p>
      <a
        className="rest-link"
        href="https://acmeoyster.com/metairie/"
        rel="noopener noreferrer"
        target="_blank"
      >
        ACME OYSTER HOUSE
      </a>
    </div>
  </div>,
  <div
    className="individual-restaurant"
    key="1"
  >
    <img
      src="https://resizer.otstatic.com/v2/photos/wide-huge/24439769.jpg"
    />
    <h5
      className="rest-name"
    >
      Zea Rotisserie & Bar
    </h5>
    <p
      className="rest-tags"
    >
      #
      eclectic #beer #mediterranean #teas
    </p>
    <div
      className="restaurant-info"
    >
      <p
        className="rest-lines"
      >
        <span
          className="bold"
        >
          Rating:
        </span>
         
        4.5
         / 5.0
      </p>
      <p
        className="rest-lines"
      >
        <span
          className="bold"
        >
          Distance:
        </span>
         
        1.7 miles
      </p>
      <p
        className="rest-lines"
      >
        <span
          className="bold"
        >
          Walk-time:
        </span>
         
        35 minutes
      </p>
      <a
        className="rest-link"
        href="http://zearestaurants.com/locations/metairie/"
        rel="noopener noreferrer"
        target="_blank"
      >
        ZEA ROTISSERIE & BAR
      </a>
    </div>
  </div>,
]
`;
